src/dsp_tools/models/connection.py

```diff
@@ -27,240 +29,260 @@ def check_for_api_error(response: requests.Response) -> None:
         )
 
 
+@dataclass
 class Connection:
     """
-    An Connection instance represents a connection to a DSP server.
-
-    Attributes
-    ----------
-
-    none (internal use attributes should not be modified/set directly)
+    A Connection instance represents a connection to a DSP server.
+
+    Attributes:
+        server: address of the server, e.g https://api.dasch.swiss
+        prefixes: ontology prefixes used
+        token: session token received by the server after login
+        log: if True, every request is written into a file
+        log_directory: directory where the log files are written
     """
 
-    _server: str
-    _prefixes: Union[dict[str, str], None]
-    _token: Union[str, None]
-    _log: bool
-
-    def __init__(self, server: str, prefixes: dict[str, str] = None):
-        """
-        Constructor requiring the server address, the user and password of DSP
-        :param server: Address of the server, e.g https://api.dasch.swiss
-        :param prefixes: Ontology prefixes used
-        """
-
-        self._server = regex.sub(r"\/$", "", server)
-        self._prefixes = prefixes
-        self._token = None
-        self._log = False
+    server: str
+    prefixes: Optional[dict[str, str]] = None
+    token: Optional[str] = None
+    log: bool = False
+    log_directory: Path = Path("HTTP requests")
 
     def login(self, email: str, password: str) -> None:
```

Comment:
Is there a use case for a non-logged-in connection? Otherwise this could be moved to the `__post_init__()` method so it doesn't conflict with immutability

Reply:
In the codebase, there is no example of a non-logged-in connection. But are there use cases where this should be possible? Should it perhaps be possible to get a project definition from a server with the `get` command, as UnknownUser?

On the other hand, that would be a privacy issue, because the users of a project are also retrieved by `get`.

Reply:
This is in fact something I've been confused about before, that get requires credentials (I thought I even opened an issue for it at some point).  
And, yes, that is potentially a privacy issue, but it's API side: It's not the client's fault to retrieve information that the API exposes publicly.

But for that use case, maybe this can just be a Connection where the token is `None`? Then the `GET` command could simply not provide the bearer header in this case.

Reply:
I tried to get rosetta from the test server, without credentials, and when retrieving the users, I get `dsp.errors.ForbiddenException: SystemAdmin or ProjectAdmin permissions are required`. If I catch this error, the rest of the project (ontologies, groups, lists, ...) can be retrieved correctly.

I think I should add support for this (would be a new feature: get a project without login)

Reply:
ah, all the better! Yes, that would be perfect. I suppose the users step should just be skipped if no credentials are provided